sklearn/utils/__init__.py

```diff
@@ -268,7 +272,13 @@ def _safe_indexing_row(X, indices):
         else:
             return X[indices]
     else:
-        return [X[idx] for idx in indices]
+        # In the case of a slice or a scalar
```

Comment:
This belongs inside the subsequent if case. And that if case could be an elif in the outer scope